### ISSUE TITLE
CommonMark loose list compatibility

### ISSUE THREAD

**Issue by waahto** [typora/typora-issues#140]
http://spec.commonmark.org/0.24/#loose
> A list is loose if any of its constituent list items are separated by blank lines, or if any of its constituent list items directly contain two block-level elements with a blank line between them. Otherwise a list is tight. (The difference in HTML output is that paragraphs in a loose list are wrapped in \<p> tags, while paragraphs in a tight list are not.)

```md
* this
* is
* a
* tight
* list

---

* this

* is

* a

* loose

* list
```

Loose list makes things like this possible:

```md
* this

  loose

* list

  has

* three list items

  each with two paragraphs
````

It seems that this can be done in source code mode but perhaps we could achive it in the normal mode also? Maybe by starting the next line on the list with two spaces would convert it from a `<li>` to `<p>` element?

**Comment by abnerlee** [comment]
> Maybe by starting the next line on the list with two spaces would convert it from a `<li>` to `<p>` element?

Well, this is a bit hacky to users and typora want to make things easier not more complicate.  Currently, the conversion between loose list and nest list is handled by typora according to list content automatically. And user shouldn't care this, if they have to care about the generated markdown source, then they need to switch to source mode. Though we do not encourage users to use source mode, but it is still an option for them in some cases.

In hybrid mode, if  a list item contains more than one paragraph, or cannot be presented as a nest list, then the list will be a loose list.

In other words, if you input 
```
* this

  loose

* list

  has

* three list items

  each with two paragraphs
```

in typora, typora will treat it as a loose list.

And you could edit the in source mode, if you want to override typora's decisions on whether a list is loose or not.
